python/src/aiconfig/editor/client/src/components/EditorContainer.tsx

```diff
@@ -665,9 +669,45 @@ export default function EditorContainer({
     return () => window.removeEventListener("keydown", saveHandler);
   }, [onSave]);
 
+  // Server heartbeat, check every 3s to show error if server is down
+  const getServerStatusCallback = callbacks.getServerStatus;
+  useEffect(() => {
+    if (!getServerStatusCallback) {
+      return;
+    }
+
+    const interval = setInterval(async () => {
+      try {
+        const res = await getServerStatusCallback();
+        setServerStatus(res.status);
+      } catch (err: unknown) {
+        setServerStatus("ERROR");
```

Comment:
Should we propagate the error message to display in the alert?

Reply:
There isn't really an error message -- it's either "OK" or not, since it will only error if the server is not able to return a response